.github/workflows/test.yml

```diff
@@ -0,0 +1,82 @@
+name: Test
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: write
+  pull-requests: write
+
+concurrency:
+  group: test-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  snapshots:
+    name: Snapshots
+    # Don't run on main
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: '3.0'
+        bundler-cache: true
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 16
+        cache: 'npm'
+        cache-dependency-path: |
+          package-lock.json
+          demo/package-lock.json
+    - run: |
+        npm ci
+        cd demo && npm ci
+    - name: Build and test with Rake
+      run: |
+        bundle exec rake test:snapshots
+    - id: auto-commit
```

Comment:
This is super interesting because you're not relying on maintainers to update snapshots manually. Is that correct? All of it is done on the server, which eliminates the need for additional Docker config too. 

I guess the trade-off here is that folks may just ignore / forget to look at the visual differences and scroll past it over time? Keen to see how this plays out either way. I like this approach.

Reply:
I think a benefit here too is that we'll ensure the images are always run on the same system, so if there's subtle differences in font rendering it won't show.